csrc/mmdeploy/model/CMakeLists.txt

```diff
@@ -5,18 +5,102 @@ project(mmdeploy_model)
 set(MODEL_NAMES "directory_model")
 if (${MMDEPLOY_ZIP_MODEL})
     set(MODEL_NAMES ${MODEL_NAMES} "zip_model")
+
+    if (MSVC)
+        set(zlib zlibstatic)
+        set(LIB_PREFIX "")
+        set(LIB_SUFFIX ".lib")
+    else()
+        set(zlib z)
+        set(LIB_PREFIX "lib")
+        set(LIB_SUFFIX ".a")
+    endif()
+    set(ziplib zip)
+
+    set(zlib_name ${LIB_PREFIX}${zlib}${LIB_SUFFIX})
+    set(ziplib_name ${LIB_PREFIX}${ziplib}${LIB_SUFFIX})
+
+    include(ExternalProject)
+    set(ZLIB_BUILD_DIR ${CMAKE_BINARY_DIR}/zlib-build)
+    set(ZLIB_INSTALL_DIR ${CMAKE_BINARY_DIR}/zlib-install)
+    set(ZLIB_INCLUDE_DIR ${ZLIB_INSTALL_DIR}/include)
+    set(ZLIB_LIBRARY_DIR ${ZLIB_INSTALL_DIR}/lib)
+    ExternalProject_Add(
```

Comment:
What's the motivation for building zlib from source?

Reply:
libzip require zlib and on windows we have to build them all